examples/client/Locomotion/src/context/ridePageStateContext/index.tsx

```diff
@@ -55,10 +56,13 @@ const RideStateContextContextProvider = ({ children }: { children: any }) => {
       setTerritory(t);
     }
     if (t && checkTerritory) {
-      const position = await getPosition();
-      const isInsidePoly = pointInPolygon(t, (position || DEFAULT_COORDS));
-      if (!isInsidePoly) {
-        setNotInTerritory();
+      const activeRide = await getActiveRide();
```

Comment:
you will call to activeRide network every time, why not checking if there is ride in the context?